controller/k8s/api.go

```diff
@@ -71,26 +71,79 @@ type API struct {
 	namespace         string
 }
 
-// NewAPI takes a Kubernetes client and returns an initialized API
+// InitializeAPI creates Kubernetes clients and returns an initialized API wrapper.
+func InitializeAPI(kubeConfig string, namespace string, resources ...APIResource) (*API, error) {
+	k8sClient, err := NewClientSet(kubeConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	// check for cluster-wide vs. namespace-wide access
+	clusterAccess, err := k8s.ClusterAccess(k8sClient, namespace)
+	if err != nil {
+		return nil, err
+	}
+	restrictToNamespace := ""
+	if !clusterAccess {
+		log.Warnf("Not authorized for cluster-wide access, limiting access to \"%s\" namespace", namespace)
```

Comment:
Not something to address right now, but possibly something to think about going forward as we refine the install process: 

We could potentially do authz checks individually for each namespace in the cluster and watch all namespaces that we have access to.  This would be helpful if you want a Linkerd control plane to manage a data plane that spans more than one namespace, but you don't have full cluster access.